src/Haskell/Ide/Engine/Plugin/HsImport.hs

```diff
@@ -110,48 +131,132 @@ importModule uri modName = pluginGetFile "hsimport cmd: " uri $ \origInput -> do
                     $ IdeResultOk (J.WorkspaceEdit newChanges newDocChanges)
             else return $ IdeResultOk (J.WorkspaceEdit mChanges mDocChanges)
 
+-- | Search style for Hoogle.
+-- Can be used to look either for the exact term,
+-- only the exact name or a relaxed form of the term.
+data SearchStyle
+  = Exact -- ^ If you want to match exactly the search string.
+  | ExactName -- ^ If you want to match exactly a function name.
+              -- Same as @Exact@ if the term is just a function name.
+  | Relax (T.Text -> T.Text) -- ^ Relax the search term to match even more.
```

Comment:
This is cool

Reply:
It might be sensible to move this to the Hoogle plugin. But right now, nobody else uses it, so it stays in HsImport.